lib/prepare.js

```diff
@@ -320,7 +322,7 @@ async function genRoutesFile ({ siteData: { pages }, sourceDir, pageFiles }) {
     component: ThemeLayout,
     beforeEnter: (to, from, next) => {
       import(${JSON.stringify(filePath)}).then(comp => {
```

Comment:
如果 filepath 包含 `!` 号，如：`/docs/hello!.md`，因为使用了 Webpack，`!` 前面的内容会被认为是 loader 的配置，见：[https://webpack.js.org/concepts/loaders/#inline](https://webpack.js.org/concepts/loaders/#inline)。在我的 PR 中，我使用了 [webpack-virtual-modules](https://github.com/sysgears/webpack-virtual-modules ) 解决了这个问题。

Reply:
Hmmm... webpack does not support the path containing `!`:  https://github.com/webpack/webpack/pull/6754

So we don't need to introduce `webpack-virtual-modules` to support `!` which should be handled by webpack instead of VuePress.

Reply:
哈哈，补充下我的思路。

文件：`/docs/hello!.md`

我们可以在 VuePress 里读取 `hello!.md` 的内容，并用它生成一个与 `hello!.md` 同级的 virtual module，比如：`docs/page-1.md`，如此即可避免 `!` 的问题，同时因为是同级目录，原文件里的路径都不用改变。

Reply:
我再用中文重申一句：我们没有必要为了一个并不合法的极少数用例来引入庞大的代码hack。

Reply:
好吧，我想 `!` 仅仅是对于 `Webpack` 不合法。

Reply:
既然 VuePress 基于 Webpack，本着职责清晰的原则，我们不应该在 VuePress 中来解决 Webpack 的问题，万一哪天 Webpack 有自己的解决方案呢？此外，你这种解决方案太复杂了，更加不应该写在这里。

如果你需要这个 feature，请给 Webpack 提 Issue 或者 PR，谢谢。